src/main/java/com/puppycrawl/tools/checkstyle/checks/modifier/ClassMemberImpliedModifierCheck.java

```diff
@@ -125,19 +139,25 @@
 
     /**
      * Control whether to enforce that {@code static} is explicitly coded
-     * on nested enums in classes.
+     * on nested enums in classes and records.
      */
     private boolean violateImpliedStaticOnNestedEnum = true;
 
     /**
      * Control whether to enforce that {@code static} is explicitly coded
-     * on nested interfaces in classes.
+     * on nested interfaces in classes and records.
```

Comment:
why "record" affect this ?

Reply:
explained at https://github.com/checkstyle/checkstyle/pull/8578#discussion_r472288927